src/gui/base/Icon.ts

```diff
@@ -40,10 +47,30 @@ export class Icon implements Component<IconAttrs> {
 				style: this.getStyle(vnode.attrs.style ?? null),
 			},
 			m.trust(icon),
-			vnode.attrs.hoverText ? m("span.tooltiptext.no-wrap", vnode.attrs.hoverText) : null,
+			vnode.attrs.hoverText &&
+				m(
+					"span.tooltiptext.no-wrap",
+					{
+						oncreate: (vnode) => {
+							const tooltip = vnode.dom as HTMLElement
+							if (this.root && tooltip) this.moveElementIfOffscreen(this.root, tooltip)
+						},
+					},
+					vnode.attrs.hoverText,
+				),
 		) // icon is typed, so we may not embed untrusted data
 	}
 
+	private moveElementIfOffscreen(root: HTMLElement, tooltip: HTMLElement): void {
+		const tooltipRect = tooltip.getBoundingClientRect()
```

Comment:
I am a bit concerned about performance. This would be executed every time we put an icon on the screen and `getBoundingClientRect` can cause layout thrashing.
Can we only do it when necessary, like on hover?

Reply:
Good idea, I've pushed the change now.